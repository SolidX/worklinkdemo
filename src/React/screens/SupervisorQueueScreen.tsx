import React, { Component } from 'react';
import AppContainer from '../common/AppBody/AppContainer';
import LeftPanel from '../common/AppBody/LeftPanel';
import LeftHeader from '../common/AppBody/LeftHeader';
import LeftBody from '../common/AppBody/LeftBody';
import RightPanel from '../common/AppBody/RightPanel';
import RightHeader from '../common/AppBody/RightHeader';
import RightBody from '../common/AppBody/RightBody';
import WorkOrderHeader from '../components/LeftPanel/Header/SuperWorkOrderHeader';
import ActionHeader from '../components/RightPanel/Header/ActionHeader';
import SuperWorkOrderList from '../components/LeftPanel/Body/SuperWorkOrderList';
import TeamList from '../components/RightPanel/Body/TeamList';
import { NavigationScreenProp } from 'react-navigation';
import { DragContainer } from '../utils/react-native-drag-drop';
import { connect } from 'react-redux';
import { Dispatch } from 'redux';
import { IState } from '../../../src/Redux';
import { ITeamMember } from '../../../src/Redux/drag-and-drop/types';
import { toggleTeamMember } from '../../Redux/drag-and-drop/actions';

export interface ISuperQueueNavProps {
  navigation: NavigationScreenProp<any, any>;
}

export interface ISuperQueueProps {
  teamMembers: ITeamMember[];
  error: string;
}
export interface ISuperQueueDispatchProps {
  onTeamMemberClicked(memberId: string): void;
}

export type Props = ISuperQueueProps &
  ISuperQueueNavProps &
  ISuperQueueDispatchProps;

export class SupervisorQueueScreen extends Component<Props> {
  state = {
    currentId: '',
    droppedInZone: null
  };

  addCurrentId = (id: string) => {
    console.log('added current Id', id);
    this.setState({ currentId: id });
  };
  removeId = () => {
    this.setState({ currentId: '' });
  };

  // toggleScroll = (visible: boolean) => {
  //   this.setState({ scrollEnabled: visible });
  // };
  componentDidMount() {
    const teamArray = this.props.teamMembers;
    console.log('teamArray', teamArray);
  }
  isDroppedInZone = () => {
    this.setState({ droppedInZone: true });
  };
  render() {
    const { teamMembers, onTeamMemberClicked, navigation } = this.props;
    const { currentId } = this.state;
    return (
      <DragContainer
        teamArray={teamMembers}
        // selectTeamMember={onTeamMemberClicked}
        isDroppedInZone={this.isDroppedInZone}
        onDragStart={() => {
          // if only one teamMember is dragged then add them to teamArray
          this.setState({ droppedInZone: false });
        }}
        onDragEnd={() => {
          this.setState({}, () => {
            // was teamMember dropped on workOrder?, if not remove from teamArray
            if (this.state.droppedInZone == false) {
              onTeamMemberClicked(currentId);
              this.removeId();
            } else {
              this.removeId();
            }
          });
        }}
      >
        <AppContainer>
          {/* ========
             LEFT
        ======== */}
          <LeftPanel>
            <LeftHeader>
              <WorkOrderHeader>My HVAC Queue</WorkOrderHeader>
            </LeftHeader>
            <LeftBody>
              <SuperWorkOrderList
                navigation={navigation}
                droppedInZone={this.isDroppedInZone}
              />
            </LeftBody>
          </LeftPanel>
          {/* ========
             RIGHT
        ======== */}
          <RightPanel>
            <RightHeader>
              <ActionHeader>HVAC Team</ActionHeader>
            </RightHeader>
            <RightBody>
              <TeamList
                selectTeamMember={onTeamMemberClicked}
                teamArray={teamMembers}
                addCurrentId={this.addCurrentId}
                currentId={this.state.currentId}
              />
            </RightBody>
          </RightPanel>
        </AppContainer>
      </DragContainer>
    );
  }
}

const mapStateToProps = (state: IState) => {
  return { dragDropState: state.dragDrop };
};
const mapDispatchToProps = (dispatch: Dispatch): ISuperQueueDispatchProps => ({
  onTeamMemberClicked: (memberId: string) =>
    dispatch(toggleTeamMember(memberId))
});
export default connect(
  mapStateToProps,
  mapDispatchToProps
)(SupervisorQueueScreen);
