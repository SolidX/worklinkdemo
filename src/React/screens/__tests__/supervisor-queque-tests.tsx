import { shallow, configure } from 'enzyme';
import React from 'react';
import configureStore from 'redux-mock-store';
import toJson from 'enzyme-to-json';
import {} from 'ts-jest';
import renderer from 'react-test-renderer';
import Props from '../SupervisorQueueScreen';
// imported as an unconnected component!
import SupervisorQueueScreen from '../SupervisorQueueScreen';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });
const mockStore = configureStore();
describe('SupervisorQueueScreen', () => {
  const navigation = {
    navigate: jest.fn(),
    dispatch: jest.fn(),
    state: {},
    goBack: jest.fn(),
    dismiss: jest.fn(),
    openDrawer: jest.fn(),
    closeDrawer: jest.fn(),
    toggleDrawer: jest.fn(),
    getParam: jest.fn(),
    setParams: jest.fn(),
    addListener: jest.fn(),
    push: jest.fn(),
    replace: jest.fn(),
    pop: jest.fn(),
    popToTop: jest.fn(),
    isFocused: jest.fn()
  };
  let wrapper: any, store;
  beforeEach(() => {
    const initialState = { teamMembers: [], error: '' };
    store = mockStore(initialState);
    // Shallow render the container passing in the mock store
    wrapper = shallow(<SupervisorQueueScreen navigation={navigation} />, {
      context: { store }
    });
  });
  it('renders correctly', () => {
    const component = wrapper.dive();
    expect(component).toMatchSnapshot();
  });
});
