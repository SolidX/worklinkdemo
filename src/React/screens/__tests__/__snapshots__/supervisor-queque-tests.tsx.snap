// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SupervisorQueueScreen renders correctly 1`] = `
<DragContainer
  isDroppedInZone={[Function]}
  onDragEnd={[Function]}
  onDragStart={[Function]}
>
  <AppContainer>
    <LeftPanel>
      <LeftHeader>
        <MyWorkOrderHeader>
          My HVAC Queue
        </MyWorkOrderHeader>
      </LeftHeader>
      <LeftBody>
        <SuperWorkOrderList
          droppedInZone={[Function]}
          navigation={
            Object {
              "addListener": [MockFunction],
              "closeDrawer": [MockFunction],
              "dismiss": [MockFunction],
              "dispatch": [MockFunction],
              "getParam": [MockFunction],
              "goBack": [MockFunction],
              "isFocused": [MockFunction],
              "navigate": [MockFunction],
              "openDrawer": [MockFunction],
              "pop": [MockFunction],
              "popToTop": [MockFunction],
              "push": [MockFunction],
              "replace": [MockFunction],
              "setParams": [MockFunction],
              "state": Object {},
              "toggleDrawer": [MockFunction],
            }
          }
        />
      </LeftBody>
    </LeftPanel>
    <RightPanel>
      <RightHeader>
        <ActionHeader>
          HVAC Team
        </ActionHeader>
      </RightHeader>
      <RightBody>
        <TeamList
          addCurrentId={[Function]}
          currentId=""
          selectTeamMember={[Function]}
        />
      </RightBody>
    </RightPanel>
  </AppContainer>
</DragContainer>
`;
